src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -75,7 +75,9 @@ internal class CSharpRecommendationService : AbstractRecommendationService
                 // Script and interactive
                 return GetSymbolsForGlobalStatementContext(context, cancellationToken);
             }
-            else if (context.IsAnyExpressionContext || context.IsStatementContext)
+            else if (context.IsAnyExpressionContext || 
+                     context.IsStatementContext || 
+                     context.SyntaxTree.IsDefiniteCastTypeContext(context.Position, context.LeftToken, cancellationToken))
```

Comment:
Add a comment?  `IsDefiniteCastTypeContext` might make people wonder why it is there.